compiler/src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -881,6 +885,10 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
     def appliedToType(targ: Type)(implicit ctx: Context): Tree =
       appliedToTypes(targ :: Nil)
 
+    /** The current tree applied to given type argument: `tree[targ]` */
+    def appliedToTypeTree(targ: Tree)(implicit ctx: Context): Tree =
```

Comment:
Move it between `appliedToTypes` and `appliedToTypeTrees`.

Reply:
Also look for `appliedToTypeTrees(xyz :: Nil)` in the compiler and replace them with `appliedToTypeTree(xyz)`. I saw some instances in `tpd` and in `QuotesAndSplices`.